### PR TITLE
Kick chargers to drop weapons

### PR DESCRIPTION
**Pull request by press-mark**
Addresses #28850 and allows players to kick weapon chargers to drop the guns out of em, pretty simple.
:cl:
 * tweak: You can now kick chargers to release any weapons inside.

### CODE REVIEW

**Comment by ShiftyRail** [comment]
What a nice buff to table-climber security

**Comment by Eneocho** [comment]
Swap it from kicking to crowbaring and we have a deal.

**Comment by bathosbathtime** [comment]
Using both methods would be best, I believe.

File: code/game/machinery/recharger.dm [hunk 1]
```diff
@@ -269,6 +269,20 @@
...
+	. = ..()
+	if(charging)
+		if(prob(25))
+			visible_message("<span class='notice'>\The [charging] pops out of \the [src]!</span>")
+			charging.appearance = appearance_backup
+			charging.update_icon()
+			charging.forceMove(get_turf(src))
+			charging = null
+		if(!self_powered)
+			use_power = 1
```
Comment (Kurfursten) [inline code/game/machinery/recharger.dm]: this should be inside the prob chance

File: code/game/machinery/recharger.dm [hunk 2]
```diff
@@ -269,6 +269,20 @@
 /obj/machinery/recharger/self_powered	//ideal for the Thunderdome
 	self_powered = 1
 
+/obj/machinery/recharger/kick_act(mob/living/carbon/human/H)
+	. = ..()
+	if(charging)
+		if(prob(25))
+			visible_message("<span class='notice'>\The [charging] pops out of \the [src]!</span>")
```
Comment (Kurfursten) [inline code/game/machinery/recharger.dm:291]: instead of pasting this, we should probably just have a proc for dropping the contained gun

**Review by Kurfursten — CHANGES_REQUESTED** [review]
those things I said

**Comment by Kurfursten** [comment]
3 week checkin

**Comment by press-mark** [comment]
> 3 week checkin

oop forgot about this, will fix it up tonight

**Comment by press-mark** [comment]
okay crowbars work on it now

**Comment by Kurfursten** [comment]
that wasn't what I asked at all!

make a proc for dropping the gun and call it from the now three (3) places that use it!

**Comment by Killette2** [comment]
We kick everything now

File: code/game/machinery/recharger.dm [hunk 3]
```diff
@@ -70,6 +70,21 @@
 				return 1
 		else
 			return 1
+	if(iscrowbar(G))
+		if(charging)
+			to_chat(usr, "You begin to pry out \the [charging].")
```
Comment (ShiftyRail) [inline code/game/machinery/recharger.dm]:
```suggestion
			to_chat(user, "You begin to pry out \the [charging].")
```

File: code/game/machinery/recharger.dm [hunk 4]
```diff
@@ -70,6 +70,21 @@
...
+				charging.appearance = appearance_backup
+				charging.update_icon()
+				charging.forceMove(loc)
+				charging = null
+				appearance_backup=null
+				G.playtoolsound(src, 50)
+				update_icon()
+				return 1
+		else
+			to_chat(usr, "You can't pry anything out of \the [src]!")
```
Comment (ShiftyRail) [inline code/game/machinery/recharger.dm]:
```suggestion
			to_chat(user, "You can't pry anything out of \the [src]!")
```

**Review by ShiftyRail — APPROVED** [review]
This is less than perfect but I don't want to live hanging for such a small change

**Pull request merged**